src/timedate/timedated.c

```diff
@@ -873,6 +873,11 @@ static int method_set_time(sd_bus_message *m, void *userdata, sd_bus_error *erro
         if (!relative && utc <= 0)
                 return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid absolute time");
 
+        /* when utc time is past systemd build date time*/
+       	/* convert epoch time to microseconds to compare with utc*/
+        if(!relative && utc <= ((int64_t)TIME_EPOCH * 1000000))
+		return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid time - past build date time");
```

Comment:
hmm, "past" is weird wording here, no? after all, we are talking about time *before*, not *after*. Hence maybe reword to "Request to set clock to time before build time, refusing". or so?

Reply:
done, made the change